src/metpy/calc/kinematics.py

```diff
@@ -378,24 +378,33 @@ def advection(
 ):
     r"""Calculate the advection of a scalar field by the wind.
 
+    If ``scalar`` is a `xarray.DataArray`, only ``u``, ``v``, and ``w`` are required
```

Comment:
This much more readable than the docstring I had put together for https://github.com/Unidata/MetPy/commit/69869d8070a0ee29ee199bc5867f5239ff30a434, thank you for cleaning it up!

One important aspect of this function, however, seems to have been glossed over in this updated docstring: it is fully flexible in what dimensions of the wind field are used (e.g., horizontal advection if `u` and `v` are provided without `w`, vertical advection if only `w` is provided, 3D if all are provided). Below is the minimal change I could come up with to refer to this capability more clearly (without being as convoluted as the prior docstring was).

```suggestion
    r"""Calculate the advection of a scalar field by 1D, 2D, or 3D winds.

    If ``scalar`` is a `xarray.DataArray`, only ``u``, ``v``, and/or ``w`` are required
```

Reply:
Thanks for bringing this up! I was aware of this capability but never put it down in words...